MeretzLib/Apple/iOS/MeretzSampleGame/MeretzSampleGame/GameViewController.m

```diff
@@ -8,351 +8,766 @@
 	(use the URL parameters "force_show=1" and "override_comingsoon=1" as needed)
 	e.g.: https://www.meretz.com/rewards/YOUR_PRODUCT_HERE?force_show=1&override_comingsoon=1
 	This will ensure you have one or more product items consumed for reporting back in the sample game here below.
 
-3) Generate a "connection code" for your test account (from step 2) and paste it into the value for BLAH
-   e.g.: #define TEST_MERETZ_USER_CONNECTION_CODE		@"def123abc456"
-
-4) At the same product page, purchase one or more available items, in order to test the /vendor/consume endpoint
-	(https://www.meretz.com/rewards/YOUR_PRODUCT_HERE?force_show=1&override_comingsoon=1)
+4) While still on your product page on the Meretz.com site, generate a "user connection code" for your test account (from step 2).
 
 5) Now build and run this sample game.
-	a) click once in the game view to initiate a /vendor/connect for your test account (watch the console for output)
-	b) assuming the /vendor/connect succeeds, click again in the game view to 
-	initiate a /vendor/consume operation for your test user (watch the console for output)
-	NOTE: a real app would want to save off the access-token returned by the VendorConnect operation for future use.
-	c) click once more in the game view to initiate a /vendor/disconnect task for your test account (watch the console for output)
-	d) that's it for current demo functionality
-
+	a) Click in the "Connection Code" text field to enter your connection code from step 4 (watch the console for output).
+	   NOTE: assuming connection is successful, a VendorConnect operation will be initiated for your test account (watch the console for output).
+ 	   NOTE: the sample game will also save the Meretz access-key for your user if the VendorConnect operation was successful.
+	   Any saved Meretz access-key will be used automatically on future program runs. Your application should follow a similar pattern.
+	b) Click the "Consume Items" button to manually check for new items purchased with Meretz points (watch the console for output).
+	d) Click the "Disconnect" button to initiate a VendorDisconnect task for your test account (watch the console for output).
+		NOTE: assuming this completes successfully, your user's account is now no longer linked with the Meretz back-end,
+		ie. the stored access-key is now invalid.
+	e) That's it for current demo functionality :)
 */
 
-#define TEST_MERETZ_USER_CONNECTION_CODE		@"def123abc456"
-#define SAMPLE_GAME_USER_IDENTIFIER				@"MeretzSampleGameUser"
+#define MERETZ_USER_ACCESS_TOKEN_NSUSERDEFAULTS_KEY			@"MeretzUserAccessToken"
+#define SAMPLE_GAME_USER_IDENTIFIER							@"MeretzSampleGameUser"
+
+// toggle this value to test delegate responders vs. manual low-level task management
+static BOOL USE_MERETZ_DELEGATE= TRUE;
 
 typedef NS_ENUM(NSInteger, MeretzSampleGameState)
 {
 	Initial,
 	Connecting,
 	Consuming,
 	Disconnecting,
-	Finished
+	Idle
 };
 
```

Comment:
Lots of improvement to the sample game:
* now has real UI
* saves the user access-key upon retrieval now
* compile time switch USE_MERETZ_DELEGATE can be toggled to demo high-level (delegate) or low-level (no delegate) implementation
* other various cleanup.